src/KubernetesClient/PeekableStreamReader.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace k8s
+{
+    public class PeekableStreamReader : IAsyncLineReader
```

Comment:
This implementation is not thread safe. Please ensure it works properly when the various methods are called at the same time on separate threads.

Reply:
I'm not sure why it needs to be thread safe?  The underlying stream that it ends up wrapped around from HttpClient (IIRC ChunkedEncodingReadStream) isn't thread safe and nor was the previous implementation of the same class.